src/components/Channel/hooks/useMentionsHandlers.ts

```diff
@@ -0,0 +1,51 @@
+import { MouseEvent, useCallback } from 'react';
+import type { UserResponse } from 'stream-chat';
+
+import type { DefaultUserType, UnknownType } from '../../../../types/types';
+
+export type OnMentionAction<Us extends UnknownType = DefaultUserType> = (
+  event: MouseEvent<HTMLElement>,
+  user?: UserResponse<Us>,
+) => void;
+
+const useMentionsHandlers = <Us extends UnknownType = DefaultUserType>(
+  onMentionsHover?: OnMentionAction<Us>,
+  onMentionsClick?: OnMentionAction<Us>,
+) =>
+  useCallback(
+    (event: MouseEvent<HTMLElement>, mentioned_users: UserResponse<Us>[]) => {
+      if (!onMentionsHover && !onMentionsClick) return;
+
+      const target = event.target;
+      // @ts-expect-error
+      const tagName = target?.tagName?.toLowerCase() as string;
+      // @ts-expect-error
+      const textContent = target?.innerHTML?.replace('*', '') as string;
```

Comment:
```suggestion
      if (!(event.target instanceof HTMLElement)) return;
      
      const target = event.target;
      const tagName = target.tagName.toLowerCase() as string;
      const textContent = target.innerHTML.replace('*', '') as string;
```

I like that approach to narrow down the target type. Throwing an error instead of `return` also works.

Reply:
I like this better as well, also no longer need the string casting